File: configurator/src/main/java/com/beezassistant/configurator/models/StarLink.java
package com.beezassistant.configurator.models;

import java.io.Serializable;

import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.ManyToOne;
import javax.persistence.MapsId;
import javax.persistence.Table;

@Entity
@Table(name="starlink")
public class StarLink implements Serializable {
	
	/**
	 * 
	 */
	private static final long serialVersionUID = -4109868076889789803L;

	@EmbeddedId
	private StarLinkId starLinkId;
	
	@ManyToOne
	@MapsId("starName")
	private Star star;

	private String linkName;
	
	public StarLink() {
		super();
		starLinkId = new StarLinkId();
	}
	
	public StarLinkId getStarLinkId() {
		return starLinkId;
	}

	public void setStarLinkId(StarLinkId starLinkId) {
		this.starLinkId = starLinkId;
	}

	

	// public String getStarName() {
	// 	return starLinkId.getStarName();
	// }

	// public void setStarName(String starName) {
	// 	starLinkId.setStarName(starName);
	// }

	public String getLink() {
		return starLinkId.getLink();
	}

	public void setLink(String link) {
		starLinkId.setLink(link);
	}

	public String getLinkName() {
		return linkName;
	}
	
	public void setLinkName(String linkName) {
		this.linkName = linkName;
	}
	
}

File: botapplicationtools/programs/programtools/starprofiletools/StarLink.py
from botapplicationtools.programs.programtools.starprofiletools.Star import Star


class StarLink:

    __star: Star
    __link: str
    __linkName: str

    def __init__(
            self,
            star: Star,
            link: str,
            linkName: str
    ):
        self.__star = star
        self.__link = link
        self.__linkName = linkName

    @property
    def getStar(self):
        return self.__star

    @property
    def getLink(self):
        return self.__link

    @property
    def getLinkName(self):
        return self.__linkName
